powertools-logging-logback/src/main/java/software/amazon/lambda/powertools/logging/LambdaEcsEncoder.java

```diff
@@ -0,0 +1,94 @@
+package software.amazon.lambda.powertools.logging;
+
+import ch.qos.logback.classic.pattern.ThrowableHandlingConverter;
+import ch.qos.logback.classic.pattern.ThrowableProxyConverter;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.ThrowableProxy;
+import ch.qos.logback.core.encoder.EncoderBase;
+import software.amazon.lambda.powertools.core.internal.LambdaHandlerProcessor;
+import software.amazon.lambda.powertools.logging.internal.LambdaEcsSerializer;
+
+import java.util.Map;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static software.amazon.lambda.powertools.logging.internal.PowertoolsLoggedFields.*;
+
+/**
+ * This class will encode the logback event into the format expected by the ECS service (ElasticSearch).
+ * <br/>
+ * Inspired from <code>co.elastic.logging.logback.EcsEncoder</code>, this class doesn't use
+ * any JSON (de)serialization library (Jackson, Gson, etc.) or Elastic library to avoid the dependency.
+ * <br/>
+ * This encoder also adds cloud information (see <a href="https://www.elastic.co/guide/en/ecs/current/ecs-cloud.html">doc</a>)
+ * and Lambda function information (see <a href="https://www.elastic.co/guide/en/ecs/current/ecs-faas.html">doc</a>, currently in beta).
+ */
+public class LambdaEcsEncoder extends EncoderBase<ILoggingEvent> {
+
+    protected static final String ECS_VERSION = "1.2.0";
+    protected static final String CLOUD_PROVIDER = "aws";
+    protected static final String CLOUD_SERVICE = "lambda";
+
+    private final ThrowableProxyConverter throwableProxyConverter = new ThrowableProxyConverter();
+    protected ThrowableHandlingConverter throwableConverter = null;
+
+    @Override
+    public byte[] headerBytes() {
+        return null;
+    }
+
+    @Override
+    public byte[] encode(ILoggingEvent event) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method calls 27 other methods. By comparison, 98% of the methods in the CodeGuru reference dataset use fewer. This indicates the method is highly coupled with other methods. A method that is highly coupled with other methods is difficult to understand and its behavior might change unexpectedly when one of its referenced methods is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 56-65, lines 68-74 into separate methods.